core/src/main/java/org/infinispan/persistence/sifs/IndexRequest.java

```diff
@@ -52,8 +52,8 @@ public static IndexRequest moved(int segment, Object key, ByteBuffer serializedK
       return new IndexRequest(Type.MOVED, segment, Objects.requireNonNull(key), serializedKey, file, offset, size, prevFile, prevOffset);
    }
 
-   public static IndexRequest dropped(int segment, Object key, ByteBuffer serializedKey, int prevFile, int prevOffset) {
-      return new IndexRequest(Type.DROPPED, segment, Objects.requireNonNull(key), serializedKey, -1, -1, -1, prevFile, prevOffset);
+   public static IndexRequest dropped(int segment, Object key, ByteBuffer serializedKey, int file, int offset, int prevFile, int prevOffset) {
```

Comment:
This change also prevented a different NPE I had found that could happen when the old file was not updating its size properly.